src/main/java/org/opentripplanner/graph_builder/module/osm/OpenStreetMapModule.java

```diff
@@ -914,33 +904,76 @@ private void buildElevatorEdges(Graph graph) {
                     // accumulate onboard vertices to so they can be connected by hop edges later
                     onboardVertices.add(onboardVertex);
                 }
+                createElevatorHopEdges(onboardVertices, node.isTagFalse("bicycle"), node.isTagFalse("wheelchair"));
+            } // END elevator edge loop
 
-                // -1 because we loop over onboardVertices two at a time
-                for (Integer i = 0, vSize = onboardVertices.size() - 1; i < vSize; i++) {
-                    Vertex from = onboardVertices.get(i);
-                    Vertex to = onboardVertices.get(i + 1);
-
-                    // default permissions: pedestrian, wheelchair, and bicycle
-                    boolean wheelchairAccessible = true;
-                    StreetTraversalPermission permission = StreetTraversalPermission.PEDESTRIAN_AND_BICYCLE;
-                    // check for bicycle=no, otherwise assume it's OK to take a bike
-                    if (node.isTagFalse("bicycle")) {
-                        permission = StreetTraversalPermission.PEDESTRIAN;
-                    }
-                    // check for wheelchair=no
-                    if (node.isTagFalse("wheelchair")) {
-                        wheelchairAccessible = false;
-                    }
+            // Add highway=elevators to graph as elevators
+            Iterator<OSMWay> elevators = osmdb.getWays().stream()
+                .filter(way -> way.hasTag("highway") && "elevator".equals(way.getTag("highway")) && !osmdb.isAreaWay(way.getId()))
+                .iterator();
+
+            for (Iterator<OSMWay> it = elevators; it.hasNext(); ) {
+                OSMWay elevatorWay = it.next();
 
-                    // The narrative won't be strictly correct, as it will show the elevator as part
-                    // of the cycling leg, but I think most cyclists will figure out that they
-                    // should really dismount.
-                    ElevatorHopEdge foreEdge = new ElevatorHopEdge(from, to, permission);
-                    ElevatorHopEdge backEdge = new ElevatorHopEdge(to, from, permission);
-                    foreEdge.wheelchairAccessible = wheelchairAccessible;
-                    backEdge.wheelchairAccessible = wheelchairAccessible;
+                List<Long> nodes = Arrays.stream(elevatorWay.getNodeRefs().toArray())
+                    .filter(nodeRef -> intersectionNodes.containsKey(nodeRef) && intersectionNodes.get(nodeRef) != null)
+                    .boxed().collect(Collectors.toList());
+
+                ArrayList<Vertex> onboardVertices = new ArrayList<Vertex>();
+                for (int i = 0; i < nodes.size(); i++) {
+                    Long node = nodes.get(i);
+                    OsmVertex sourceVertex = intersectionNodes.get(node);
+                    String sourceVertexLabel = sourceVertex.getLabel();
+
+                    ElevatorOffboardVertex offboardVertex = new ElevatorOffboardVertex(
```

Comment:
This is almost identical to the code for the vertical elevators. Is it possible to extract a reusable method?